clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,398 @@
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{fs, path::PathBuf};
+    use tracing::level_filters::LevelFilter;
+    use uuid::Uuid;
+
+    use super::{update_config, MqttConfig, TeliodDaemonConfig};
+    use crate::configure_interface::InterfaceConfigurationProvider;
+
+    pub const TELIOD_CFG: &str = "/tmp/teliod_config.json";
+
+    #[test]
+    fn test_update_config() {
+        let initial_config = r#"
+        {
+            "log_level": "debug",
+            "log_file_path": "/path/to/log",
+            "authentication_token": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            "app_user_uid": "00000000-0000-0000-0000-000000000000",
+            "interface": {
+                "name": "eth0",
+                "config_provider": "manual"
+            }
+        }
+        "#;
+        fs::write(TELIOD_CFG, initial_config).unwrap();
+
+        let update_body = r#"
+        {
+            "log_level": "info",
+            "log_file_path": "/new/path/to/log",
+            "authentication_token": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+            "app_user_uid": "11111111-1111-1111-1111-111111111111",
+            "interface": {
+                "name": "eth1",
+                "config_provider": "ifconfig"
+            }
+        }
+        "#;
+
+        let response = update_config(update_body);
+        assert_eq!(response.status(), 200);
+
+        let updated_config: TeliodDaemonConfig =
+            serde_json::from_str(&fs::read_to_string(TELIOD_CFG).unwrap()).unwrap();
+
+        assert_eq!(
+            updated_config.log_level,
+            tracing::level_filters::LevelFilter::INFO
+        );
+        assert_eq!(updated_config.log_file_path, "/new/path/to/log");
+        assert_eq!(
+            updated_config.authentication_token,
+            "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+        );
+        assert_eq!(
+            updated_config.app_user_uid,
+            Uuid::parse_str("11111111-1111-1111-1111-111111111111").unwrap()
+        );
+        assert_eq!(updated_config.interface.name, "eth1");
+        assert_eq!(
+            updated_config.interface.config_provider,
+            InterfaceConfigurationProvider::Ifconfig
+        );
+        assert_eq!(updated_config.http_certificate_file_path, None);
+        let mqtt_default_cfg = MqttConfig::default();
+        assert_eq!(
+            updated_config.mqtt.backoff_initial,
+            mqtt_default_cfg.backoff_initial
+        );
+        assert_eq!(
+            updated_config.mqtt.backoff_maximal,
+            mqtt_default_cfg.backoff_maximal
+        );
+        assert_eq!(
+            updated_config.mqtt.reconnect_after_expiry,
+            mqtt_default_cfg.reconnect_after_expiry
+        );
+        assert_eq!(
+            updated_config.mqtt.certificate_file_path,
+            mqtt_default_cfg.certificate_file_path
+        );
```

Comment:
nit: I think it would be more readable if you would create the expected struct and compare it against the `updated_config`, instead of manually doing it field by field.

Reply:
Agreed, changed